x/mint/keeper/keeper_test.go

```diff
@@ -60,109 +60,130 @@ func TestKeeperTestSuite(t *testing.T) {
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.Staking).TruncateInt(),
-		suite.App.BankKeeper.GetAllBalances(suite.Ctx, feeCollector).AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.CommunityPool),
-		feePool.CommunityPool.AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.DeveloperRewards).TruncateInt(),
-		suite.App.BankKeeper.GetBalance(suite.Ctx, devRewardsReceiver, "stake").Amount)
+		})
+	}
+}
 
-	// Test for multiple dev reward addresses
-	params.WeightedDeveloperRewardsReceivers = []types.WeightedAddress{
+func (suite *KeeperTestSuite) TestDistrAssetToDeveloperRewardsAddrWhenNotEmpty() {
+	var (
+		distrTo = lockuptypes.QueryCondition{
+			LockQueryType: lockuptypes.ByDuration,
+			Denom:         "lptoken",
+			Duration:      time.Second,
+		}
+		params       = suite.App.MintKeeper.GetParams(suite.Ctx)
+		gaugeCoins   = sdk.Coins{sdk.NewInt64Coin("stake", 10000)}
+		gaugeCreator = sdk.AccAddress([]byte("addr2---------------"))
+		mintLPtokens = sdk.Coins{sdk.NewInt64Coin(distrTo.Denom, 200)}
+	)
+
+	tests := []struct {
+		name              string
+		weightedAddresses []types.WeightedAddress
+		mintCoin          sdk.Coin
+	}{
 		{
-			Address: devRewardsReceiver2.String(),
-			Weight:  sdk.NewDecWithPrec(6, 1),
+			name: "one dev reward address",
+			weightedAddresses: []types.WeightedAddress{
+				{
+					Address: sdk.AccAddress([]byte("addr1---------------")).String(),
+					Weight:  sdk.NewDec(1),
+				}},
+			mintCoin: sdk.NewCoin("stake", sdk.NewInt(10000)),
 		},
 		{
-			Address: devRewardsReceiver3.String(),
-			Weight:  sdk.NewDecWithPrec(4, 1),
+			name: "multiple dev reward addresses",
+			weightedAddresses: []types.WeightedAddress{
+				{
+					Address: sdk.AccAddress([]byte("addr3---------------")).String(),
+					Weight:  sdk.NewDecWithPrec(6, 1),
+				},
+				{
+					Address: sdk.AccAddress([]byte("addr4---------------")).String(),
+					Weight:  sdk.NewDecWithPrec(4, 1),
+				}},
+			mintCoin: sdk.NewCoin("stake", sdk.NewInt(100000)),
 		},
 	}
-	suite.App.MintKeeper.SetParams(suite.Ctx, params)
-
-	err = mintKeeper.MintCoins(suite.Ctx, mintCoins)
-	suite.NoError(err)
-	err = mintKeeper.DistributeMintedCoin(suite.Ctx, mintCoin)
-	suite.NoError(err)
-
-	suite.Equal(
-		mintCoins[0].Amount.ToDec().Mul(params.DistributionProportions.DeveloperRewards).Mul(params.WeightedDeveloperRewardsReceivers[0].Weight).TruncateInt(),
-		suite.App.BankKeeper.GetBalance(suite.Ctx, devRewardsReceiver2, "stake").Amount)
-	suite.Equal(
-		mintCoins[0].Amount.ToDec().Mul(params.DistributionProportions.DeveloperRewards).Mul(params.WeightedDeveloperRewardsReceivers[1].Weight).TruncateInt(),
-		suite.App.BankKeeper.GetBalance(suite.Ctx, devRewardsReceiver3, "stake").Amount)
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			suite.Setup()
```

Comment:
The test feels a little verbose due to accessing keepers. For all keepers, consider having:
```
mintKeeper := suite.App.MintKeeper
```

Then, we could do `mintKeeper.SetParams(...)` instead of `suite.App.MintKeeper.SetParams(...)` , as a result, reducing verbosity and improving readability